pex/resolve/lockfile/__init__.py

```diff
@@ -77,6 +84,51 @@ def store(
         json.dump(json_codec.as_json_data(lockfile), fp, sort_keys=True)
 
 
+class CreateLockDownloadManager(DownloadManager):
+    @classmethod
+    def create(
+        cls,
+        download_dir,  # type: str
+        downloaded,  # type: Downloaded
+        pex_root=None,  # type: Optional[str]
+    ):
+        # type: (...) -> CreateLockDownloadManager
+
+        artifacts_by_filename = {
+            artifact.filename: artifact
+            for locked_resolve in downloaded.locked_resolves
+            for locked_requirement in locked_resolve.locked_requirements
+            for artifact in locked_requirement.iter_artifacts()
+        }
+        path_by_artifact = {
+            artifacts_by_filename[f]: os.path.join(root, f)
```

Comment:
Are we certain that every file we encounter in the filesystem walk is an artifact? No other crap can end up in there? I.e., is it worth checking that the `artifacts_by_filename[f]` doesn't blow up?

Reply:
We're certain. This class should probably be named `_CreateLockDownloadManager` to underscore its local-only private use, but down here we set up the empty temporary download dir: https://github.com/pantsbuild/pex/pull/1650/files#diff-eda2c2a78c6edeac766676b01baf61449994f90376ca6d9feb099e2817edaeceR175 and the Pip we vendor only plops distributions from our call to `pip download --dest ...` there.